packages/office-ui-fabric-react/src/components/SelectedItemsList/SelectedPeopleList/Items/EditingItem.tsx

```diff
@@ -1,38 +1,76 @@
 import { IPersonaProps } from '../../../../Persona';
 
-import * as stylesImport from './EditingItem.scss';
-
-// tslint:disable-next-line:no-any
-const styles: any = stylesImport;
-
-export interface IEditingSelectedPeopleItemProps extends ISelectedPeopleItemProps {
-  // tslint:disable-next-line:no-any
-  onEditingComplete: (oldItem: any, newItem: any) => void;
-  onRenderFloatingPicker?: React.ComponentType<IBaseFloatingPickerProps<IPersonaProps>>;
+import * as styles from './EditingItem.scss';
+
+export interface IEditingItemProps<TItem> extends React.HTMLAttributes<any> {
+  /**
+   * The current item of the EditingItem
+   */
+  item: TItem;
+
+  /**
+   * Callback for when the edited item's new value has been selected.
+   * Invoked indirectly by the picker mounted by onRenderFloatingPicker.
+   */
+  onEditingComplete: (oldItem: TItem, newItem: TItem) => void;
+
+  /**
+   * Renders the floating picker for suggesting the result of the item edit.
+   *
+   * Not actually optional, since is what is needed to resolve the new item.
+   */
+  onRenderFloatingPicker?: React.ComponentType<EditingItemFloatingPickerProps<IPersonaProps>>;
+
+  /**
+   * Callback for when the editing item removes the item from the well
+   *
+   * Called when the item is currently being edited and the text length goes to zero
+   */
+  onRemoveItem?: (item: TItem) => void;
+
+  /**
+   * Custom props to be passed to the floating picker opened when editing the pill.
+   *
+   * @deprecated Instead of using this prop, bind this yourself inyour onRenderFloatingPicker.
+   */
   floatingPickerProps?: IBaseFloatingPickerProps<IPersonaProps>;
+
+  /**
+   * Callback used by the EditingItem to populate the initial thing
+   *
+   * Not actually optional
+   */
   getEditingItemText?: (item: IExtendedPersonaProps) => string;
 }
 
-export class EditingItem extends BaseComponent<IEditingSelectedPeopleItemProps, IPeoplePickerItemState> {
+export type EditingItemFloatingPickerProps<T> = Pick<
+  IBaseFloatingPickerProps<T>,
+  'componentRef' | 'onChange' | 'inputElement' | 'selectedItems' | 'onRemoveSuggestion'
+>;
+
+/**
+ * Wrapper around an item in a selection well that renders an item with a context menu for
+ * replacing that item with another item.
+ */
+export class EditingItem<TItem> extends React.PureComponent<IEditingItemProps<TItem>, IPeoplePickerItemState> {
   private _editingInput: HTMLInputElement;
   private _editingFloatingPicker = React.createRef<FloatingPeoplePicker>();
 
-  constructor(props: IEditingSelectedPeopleItemProps) {
+  constructor(props: IEditingItemProps<TItem>) {
     super(props);
     this.state = { contextualMenuVisible: false };
   }
 
   public componentDidMount(): void {
-    const getEditingItemText = this.props.getEditingItemText as (item: IExtendedPersonaProps) => string;
-    const itemText = getEditingItemText(this.props.item);
+    // TODO remove this cast and make the item required in types.
```

Comment:
Is this meant to be done as part of this PR?

Reply:
I see a thumbs up which I interpreted as needing resolution, but I still see the TODO. Was it meant to be resolved? 😄

Reply:
cc @Adjective-Object

Reply:
I definitely did this in https://github.com/OfficeDev/office-ui-fabric-react/blame/3551e64e93e56e7df1488041b7695ab75a97ebc5/packages/office-ui-fabric-react/src/components/FloatingPicker/BaseFloatingPicker.tsx#L271, but I'm not seeing it my local changes or in diff view. I think I may be in a bad state

Reply:
oh the TODO comment specifically I forgot to remove 
![image](https://user-images.githubusercontent.com/1174858/55363611-7962fe80-5492-11e9-8d03-54aeef107e8e.png)

Reply:
@Adjective-Object were you planning to commit additional changes? Asking to understand the status of this PR :).